### ISSUE TITLE
header guards?

### ISSUE THREAD

**Issue by anticrisis** [sdilts/cl-bindgen#7]
Hi there, I'm just trying out your library, I wonder if this is expected behaviour with respect to the include guard at the top of the file? Many thanks!

```sh
(venv) anticrisis@Au14-1:/tmp/test$ cat cpp.h
#ifndef CPP_H
#define CPP_H

#ifdef __cplusplus
extern "C" {
#endif

  int hello();

#ifdef __cplusplus
}
#endif
#endif
(venv) anticrisis@Au14-1:/tmp/test$ cl-bindgen f cpp.h
Found macro +cpp-h+ definition in: cpp.h:2:9

;; next section imported from file cpp.h

#| MACRO_DEFINITION
(defconstant +cpp-h+ ACTUAL_VALUE_HERE)
|#

(cffi:defcfun "hello" :int)
(venv) anticrisis@Au14-1:/tmp/test$
```

**Comment by sdilts** [comment]
Ya, that's the expected behavior right now. One of the challenges with automatically generating bindings from C is that code often uses `#define` to define constants used in the program instead of using `const` variables. Debate on if you should actually do that aside, it's something that the library needs to support.

The library just dumps anything that's been established using `#define` to the file. It's [probably possible to extract the whole macro definition](https://stackoverflow.com/a/17373351/4458609), but it's not something I figured out when writing the code. To get around not having a value to populate the constant with, the app just dumps out a commented out definition of the macro and lets you choose to populate it with the correct values or delete it.

If we can get the macro values, we could just exclude anything without a value, which would solve this problem.

**Comment by sdilts** [comment]
I just merged in a feature (29da42c710d3acb208dc8004768bd9d7a3b886ac) that should take care of this. It uses the name of the macro as well as the name of the file to determine if a macro is a header guard. The function it uses to do the check is  configurable, so if your code does something weird that the default implementation doesn't handle, users can write their own checking functions.

**Comment by sdilts** [comment]
Released in `1.3.0`. Let me know if you experience any problems.
